packages/opentelemetry-instrumentation-vertexai/opentelemetry/instrumentation/vertexai/__init__.py

```diff
@@ -1 +1,263 @@
 """OpenTelemetry Vertex AI instrumentation"""
+import logging
+import os
+import types
+from typing import Collection
+from wrapt import wrap_function_wrapper
+
+from opentelemetry import context as context_api
+from opentelemetry.trace import get_tracer, SpanKind
+from opentelemetry.trace.status import Status, StatusCode
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import (
+    _SUPPRESS_INSTRUMENTATION_KEY,
+    unwrap
+)
+
+from opentelemetry.semconv.ai import SpanAttributes, LLMRequestTypeValues
+from opentelemetry.instrumentation.vertexai.version import __version__
+
+from vertexai.generative_models._generative_models import Part
+
+logger = logging.getLogger(__name__)
+
+_instruments = ("google-cloud-aiplatform",)
+
+WRAPPED_METHODS = [
+    {
+        "package": "vertexai.preview.generative_models",
+        "object": "GenerativeModel",
+        "method": "generate_content",
+        "span_name": "vertexai.generate_content",
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict",
+        "span_name": "vertexai.predict",
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_async",
+        "span_name": "vertexai.predict",
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming",
+        "span_name": "vertexai.predict",
+    },
+    {
+        "package": "vertexai.language_models",
+        "object": "TextGenerationModel",
+        "method": "predict_streaming_async",
+        "span_name": "vertexai.predict",
+    },
+]
+
+
+def should_send_prompts():
+    return (
+        os.getenv("TRACELOOP_TRACE_CONTENT") or "true"
+    ).lower() == "true" or context_api.get_value("override_enable_content_tracing")
+
+def is_streaming_response(response):
+    return isinstance(response, types.GeneratorType)
+
+def is_async_streaming_response(response):
+    return isinstance(response, types.AsyncGeneratorType)
+
+def _set_span_attribute(span, name, value):
+    if value is not None:
+        if value != "":
+            span.set_attribute(name, value)
+    return
+
+def _set_input_attributes(span, args, kwargs):
+
+    if (should_send_prompts() and args is not None and len(args) > 0):
+        prompt = ''
+        for arg in args:
+            if isinstance(arg, str):
+                prompt = f"{prompt}{arg}\n"
+            elif isinstance(arg, list):
+                for subarg in arg:
+                    prompt = f"{prompt}{subarg}\n"
+
+        _set_span_attribute(
+            span,
+            f"{SpanAttributes.LLM_PROMPTS}.0.user",
+            prompt,
+        )
+
+    _set_span_attribute(span, SpanAttributes.LLM_REQUEST_MODEL, 'unknown')
```

Comment:
We should track another wrap_method: `from_trained`  to get the model name

Reply:
How will you correlate it with the call to predict?

Reply:
Thats the reason I maintained some global context so that model name can be set using from_trained and set value to all other methods span attribute values.

Reply:
But that won't work if the user has multiple instances

Reply:
Can you implement your suggestion here?